### PR TITLE
docs: improve structure

### PR DESCRIPTION
**Pull request by hsjobeki**
### Would be happy to merge :+1: 

- Renamed: architectural-considerations -> architecture.
  - also now only containing the current `architecture`.
  - considerations that led to this architecture moved to `intro` 
- created empty draft sections, for the different api modules. (fetchers, indexers, builders, translators)
  - to be filled later on. -> everyone can make a PR to document one sub-module with little effort.
  - generic-lock is already well described. (moved from the fetchers into its own section)
  - added a short missing intro sentence for fetchers.
  
 See Preview screenshot below:
 
image

### CODE REVIEW

File: docs/src/api/generic-lock.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# Generic lockfile
```
Comment (hsjobeki) [inline docs/src/api/generic-lock.md:1]: moved here from `fetchers.md`

File: docs/src/intro.md [hunk 2]
```diff
@@ -22,9 +22,60 @@ The goal of this project is to create a standardized, generic, modular framework
 
 The intention is to integrate many existing 2nix converters into this framework, thereby improving many of the previously named aspects and providing a unified UX for all 2nix solutions.
 
+
+### Modularity:
```
Comment (hsjobeki) [inline docs/src/intro.md:26]: moved here from `architectural-considerations.md` also fits nicely with the sections above.

File: docs/src/intro/fetchers.md [hunk 3]
```diff
@@ -1,38 +1,6 @@
...
-    "hash": "sha512-xrO3+1bxSo3ZVHAnqzyuewYT6aMFHRAd4Kcs92MAonjwQZLsK9d0SF1IyQ3k5PoirxTW0Oe/RqFgMQ6TcNE5Ew==",
-    "type": "http",
-    "url": "https://registry.npmjs.org/fast-glob/-/fast-glob-3.2.11.tgz"
-  }
-}
-```
-
-The `type` field tells dream2nix what fetchers to invoke to get this
-source, the rest of the attributes are simply passed to the fetcher.
+This section describes the available source code fetchers. A fetcher is responsible for loading sources defined in the generic lockfile described earlier.
```
Comment (hsjobeki) [inline docs/src/intro/fetchers.md:3]: as the lockfile description is removed; An introducing sentence is now missing. Let me know if i need to improve that.

File: docs/src/SUMMARY.md [hunk 4]
```diff
@@ -12,13 +12,35 @@
...
+  - [path]()
+  - [npm]()
+  - [pypi-sdist]()
+  - [crates-io]()
+- [Builders]()
+  - [nodejs]()
+- [Override system](./intro/override-system.md)
+
+# Correlation with nixpkgs
+- [Nixpkgs improvements](./intro/nixpkgs-improvements.md)
```
Comment (DavHau) [inline docs/src/SUMMARY.md:43]: As soon as our refactoring to make use of the nixos modules is finished, we should have automatically generated documentation for the individual modules. Therefore, I think we can remove the placeholders.

Comment (hsjobeki) [reply docs/src/SUMMARY.md]: removed the placeholders.

**Comment by DavHau** [comment]
Thanks a lot. Improvements on the docs are very important for this project.

**Pull request merged**